core/src/main/java/org/apache/iceberg/rest/Endpoint.java

```diff
@@ -0,0 +1,146 @@
+ * resource path as defined in the Iceberg OpenAPI REST specification without parameter
+ * substitution, such as <b>/v1/{prefix}/namespaces/{namespace}</b>.
+ */
+public class Endpoint {
+
+  // namespace endpoints
+  public static final Endpoint V1_LIST_NAMESPACES =
+      Endpoint.create("GET", ResourcePaths.V1_NAMESPACES);
+  public static final Endpoint V1_LOAD_NAMESPACE =
+      Endpoint.create("GET", ResourcePaths.V1_NAMESPACE);
+  public static final Endpoint V1_CREATE_NAMESPACE =
+      Endpoint.create("POST", ResourcePaths.V1_NAMESPACES);
+  public static final Endpoint V1_UPDATE_NAMESPACE =
+      Endpoint.create("POST", ResourcePaths.V1_NAMESPACE_PROPERTIES);
+  public static final Endpoint V1_DELETE_NAMESPACE =
+      Endpoint.create("DELETE", ResourcePaths.V1_NAMESPACE);
+  public static final Endpoint V1_COMMIT_TRANSACTION =
+      Endpoint.create("POST", ResourcePaths.V1_TRANSACTIONS_COMMIT);
+
+  // table endpoints
+  public static final Endpoint V1_LIST_TABLES = Endpoint.create("GET", ResourcePaths.V1_TABLES);
+  public static final Endpoint V1_LOAD_TABLE = Endpoint.create("GET", ResourcePaths.V1_TABLE);
+  public static final Endpoint V1_CREATE_TABLE = Endpoint.create("POST", ResourcePaths.V1_TABLES);
+  public static final Endpoint V1_UPDATE_TABLE = Endpoint.create("POST", ResourcePaths.V1_TABLE);
+  public static final Endpoint V1_DELETE_TABLE = Endpoint.create("DELETE", ResourcePaths.V1_TABLE);
+  public static final Endpoint V1_RENAME_TABLE =
+      Endpoint.create("POST", ResourcePaths.V1_TABLE_RENAME);
+  public static final Endpoint V1_REGISTER_TABLE =
+      Endpoint.create("POST", ResourcePaths.V1_TABLE_REGISTER);
+  public static final Endpoint V1_REPORT_METRICS =
+      Endpoint.create("POST", ResourcePaths.V1_TABLE_METRICS);
+
+  // view endpoints
+  public static final Endpoint V1_LIST_VIEWS = Endpoint.create("GET", ResourcePaths.V1_VIEWS);
+  public static final Endpoint V1_LOAD_VIEW = Endpoint.create("GET", ResourcePaths.V1_VIEW);
+  public static final Endpoint V1_CREATE_VIEW = Endpoint.create("POST", ResourcePaths.V1_VIEWS);
+  public static final Endpoint V1_UPDATE_VIEW = Endpoint.create("POST", ResourcePaths.V1_VIEW);
+  public static final Endpoint V1_DELETE_VIEW = Endpoint.create("DELETE", ResourcePaths.V1_VIEW);
+  public static final Endpoint V1_RENAME_VIEW =
+      Endpoint.create("POST", ResourcePaths.V1_VIEW_RENAME);
+
+  private static final Splitter ENDPOINT_SPLITTER = Splitter.on(" ");
+  private static final Joiner ENDPOINT_JOINER = Joiner.on(" ");
+  private final String httpMethod;
+  private final String path;
+
+  private Endpoint(String httpMethod, String path) {
+    Preconditions.checkArgument(
+        !Strings.isNullOrEmpty(httpMethod), "Invalid HTTP method: null or empty");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(path), "Invalid path: null or empty");
+    this.httpMethod = Method.normalizedValueOf(httpMethod).toString();
```

Comment:
nit: All `Method.normalizedValueOf` is doing is a `toUpperCase` so probably not worth adding the http5 dependency.  We already do the null check, so this really isn't doing anything meaningful as opposed to just `httpMethod.toUppercase(...)`

Reply:
@danielcweeks besides the uppercasing it also makes sure that this is actually a valid HTTP method. There were some earlier comments that indicated we should make sure that `httpMethod` is valid and that's most likely the easiest way of ensuring that